### PR TITLE
Redesign collection / pagination support

### LINKED ISSUE #53: Redesign collection / pagination support
Current pagination support tends to confuse SDK users as it is not explicit enough when only 1 page is used and when all pages are processed. Simplify model so user needs to explicitly state whether he wants to get current page items or all automatically paged items without the need to know runtime class.

### PR DESCRIPTION
**Pull request by jakub-plichta**
Old paging was confusing to users as they wouldn't know whether they'd
list only current page or all pages without inspecting runtime class of
the collection. The new paging drops implementation of List interface
and introduces new methods that guide user to choose whether to list
items of the current page all from all pages.

Because this commit drops List interface from page classes it also
renames all classes related to paging.

resolves #53

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: src/main/java/com/gooddata/collections/CustomPageRequest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2004-2017, GoodData(R) Corporation. All rights reserved.
```
Comment (jimirocks) [inline src/main/java/com/gooddata/collections/CustomPageRequest.java]: Could we use 2019 since it looks like a new class?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-rest-common-merge-compliance-check/12/ : SUCCESS in 15s

**Pull request merged**